nav2_collision_monitor/src/circle.cpp

```diff
@@ -94,15 +96,77 @@ bool Circle::getParameters(
       node, polygon_name_ + ".radius", rclcpp::PARAMETER_DOUBLE);
     radius_ = node->get_parameter(polygon_name_ + ".radius").as_double();
     radius_squared_ = radius_ * radius_;
-  } catch (const std::exception & ex) {
+
+    // Do not need to proceed further, if "radius" parameter is defined.
+    // Static polygon will be used.
+    polygon_sub_topic.clear();
+    return true;
+  } catch (const rclcpp::exceptions::ParameterUninitializedException &) {
+    RCLCPP_INFO(
+      logger_,
+      "[%s]: Polygon circle radius is not defined. Using dynamic subscription instead.",
+      polygon_name_.c_str());
+  }
+
+  if (polygon_sub_topic.empty()) {
     RCLCPP_ERROR(
       logger_,
-      "[%s]: Error while getting circle parameters: %s",
-      polygon_name_.c_str(), ex.what());
+      "[%s]: Error while getting polygon parameters: static radius and sub topic both not defined",
```

Comment:
```suggestion
      "[%s]: Error while getting circle parameters: static radius and sub topic both not defined",
```
This is still sp. to circle, no?